engine/storage/image/src/main/java/org/apache/cloudstack/storage/image/TemplateServiceImpl.java

```diff
@@ -286,6 +286,23 @@ public void handleSysTemplateDownload(HypervisorType hostHyper, Long dcId) {
         }
     }
 
+    protected boolean isSkipTemplateStoreDownload(VMTemplateVO template, Long zoneId) {
+        if (template.isPublicTemplate()) {
```

Comment:
@shwstppr 
if an error template is public or featured, will it become Active ?

Reply:
@weizhouapache no.
Server won't skip download if any of the following condition is true:
- Template is public
- Template is featured
- Template belongs to system
- Template is marked for the zone but not present in any of the zone secondary store

Depending upon download status template status will be set later

Reply:
it seems I misunderstood the code :-D

is there a scenario that template download is skipped but it is expected behaviour ? direct download ?

Reply:
Direct-download templates are skipped (already) at line 496